packages/components/src/buttons.test.tsx

```diff
@@ -17,10 +15,10 @@ jest.mock('./Clickable', (): ClickableExports => {
 
 describe.each([{ component: ButtonPrimary }, { component: ButtonSecondary }, { component: ButtonOutline }])(
   '$component.name',
-  ({ component }) => {
+  ({ component: Component }) => {
```

Comment:
Where is the `Component` definition coming from? I see the describe.each, but I don't see how those options are getting mapped to `Component`.

Reply:
`describe.each` calls the function with each entry in the list.
https://jestjs.io/docs/api#describeeachtablename-fn-timeout

Each entry has a `component` key. The entry is destructured and the value for that `component` key is stored in a variable named `Component`. The variable name must be capitalized because of JSX.

Reply:
I think that it is mostly the syntax that I am confused about.

`({ component: Component }) => {`

`component` seems to be coming from the `describe.each `, but `:Component` doesn't seem to be a variable but a type. I see `({ component: Component })` as saying this function takes a variable named `component` of the type Component.

Am I misreading that?

Reply:
Ah, I see. This is destructuring the first argument to the function inline. It does not have a type declaration. The type is inferred from the type of the `describe.each` table.

This is equivalent to.
```ts
(entry) => {
  const Component = entry.component;
  ...
```

If there was a type declaration, it would look something like:
```ts
({ component: Component }: { component: React.Component }) => {
  ...
}
```

But I can see how this is easily confusing, since the `Component` variable name is the same as React's `Component` type. Will rename this variable.

Reply:
Thanks for explaining

Reply:
Renamed in https://github.com/DataBiosphere/terra-ui/pull/4637/commits/f8c89448b548fac582d7be8841f34395e0522eee.